### PR TITLE
Extend the project description to 3 lines if character limit not exceeded

### PR DESCRIPTION
**Pull request by johnnyd-eth**
## What does this PR do and why?

Before: 

Screen Shot 2022-05-20 at 12 53 10 pm

After: 

Screen Shot 2022-05-20 at 1 18 33 pm

## Screenshots or screen recordings

_If applicable, provide screenshots or screen recordings to demonstrate the
changes._

## Acceptance checklist

- [ ] I have evaluated the [Approval Guidelines](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#approval-guidelines) for this PR.
- [ ] I have tested this PR in [all supported browsers](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#supported-browsers).

### CODE REVIEW

File: src/components/shared/Paragraph.tsx [hunk 1]
```diff
@@ -13,15 +13,26 @@ export default function Paragraph({
 }) {
   const CHARACTER_LIMIT_EXCEEDED =
     (characterLimit && description.length > characterLimit) ||
-    description.split('\n').length > 1
+    description.split('\n').length > 3
```
Comment (wraeth-eth) [inline src/components/shared/Paragraph.tsx:16]: One thing to possibly consider is if they have one super long line of text. I wonder if we should expand this to check each line for a character limit, wdyt?

Something like:
```ts
const isCharacterLineExceeded(description: string, characterLimit: number | undefined ) {
  if (!characterLimit) return false
  const lines = description.split('\n')
  if (lines > 3) return false
  return lines.every(line => line.length < characterLimit)
}
...
const CHARACTER_LIMIT_EXCEEDED = isCharacterLineExceeded(description, characterLimit)
```

File: src/components/shared/Paragraph.tsx [hunk 2]
```diff
@@ -13,15 +13,26 @@ export default function Paragraph({
...
 
   const [expanded, setExpanded] = useState<boolean>(false)
   const toggleExpanded = () => setExpanded(!expanded)
 
-  // truncate the first line of the description
+  const allLines = description.split('\n')
+  let firstThreeLines = ''
+
+  // Truncate the first 3 lines of the description
+  allLines.forEach((line, index) => {
```
Comment (wraeth-eth) [inline src/components/shared/Paragraph.tsx:25]: you probably only need to loop over the first three lines with slice(0, 3), and then could join them with a new line:
```ts
const firstThreeLines = allLines
  .slice(0, 3)
  .join('\n')
```

wdyt?

Comment (wraeth-eth) [reply src/components/shared/Paragraph.tsx:25]: You could even possibly put all of this into the memo itself so it only runs when the character limit, or the description changes

**Review by wraeth-eth — CHANGES_REQUESTED** [review]
Looking good - good features for sure! A bit tired at the moment so hopefully my suggestions make sense, lmk what you think

File: src/components/shared/Paragraph.tsx (hunk 1 above)
Comment (johnnyd-eth) [reply src/components/shared/Paragraph.tsx:16]: @wraeth-eth Not sure I follow this. If one line is super long then it'll just get cut off by the characterLimit right?

File: src/components/shared/Paragraph.tsx (hunk 2 above)
Comment (johnnyd-eth) [reply src/components/shared/Paragraph.tsx:25]: > you probably only need to loop over the first three lines with slice(0, 3), and then could join them with a new line:
> 
> ```ts
> const firstThreeLines = allLines
>   .slice(0, 3)
>   .join('\n')
> ```
> 
> wdyt?

Beautiful ser

Comment (johnnyd-eth) [reply src/components/shared/Paragraph.tsx:25]: > ```ts
> const firstThreeLines = allLines
>   .slice(0, 3)
>   .join('\n')
> ```

Yep another great call

File: src/components/shared/Paragraph.tsx (hunk 1 above)
Comment (wraeth-eth) [reply src/components/shared/Paragraph.tsx:16]: Yeah I think you're right - resolving

**Review by wraeth-eth — APPROVED** [review]

**Pull request merged**